backbone.js

```diff
@@ -398,10 +401,20 @@
     this.attributes = {};
     if (options.collection) this.collection = options.collection;
     if (options.parse) attrs = this.parse(attrs, options) || {};
-    attrs = _.defaults({}, attrs, _.result(this, 'defaults'));
-    this.set(attrs, options);
-    this.changed = {};
-    this.initialize.apply(this, arguments);
+    var model = this, args = slice.call(arguments);
+    var defaults = _.result(this, 'defaults');
+    var consume = function(defaults) {
+      attrs = _.defaults({}, attrs, defaults);
+      model.set(attrs, options);
+      model.changed = {};
+      var init = model.initialize.apply(model, args);
+      if (isPromise(init))
+        return init.then(_.constant(model));
+      return model;
+    };
+    if (isPromise(defaults))
+      return defaults.then(consume);
+    return consume(defaults);
```

Comment:
:-1: on this. Initialize is always synchronous, therefore this will always be resolved after you've created the instance -- in other words, exactly where it was resolved before. Revert this please.